homeassistant/components/light/__init__.py

```diff
@@ -241,7 +241,20 @@ def async_handle_light_service(service):
         if color_name is not None:
             params[ATTR_RGB_COLOR] = color_util.color_name_to_rgb(color_name)
 
+        xy_color = params.get(ATTR_XY_COLOR, None)
+
         for light in target_lights:
+            # Convert xy_color if unsupported
+            if xy_color is not None:
+                support_xy = bool(light.supported_features & SUPPORT_XY_COLOR)
+                brightness = params.get(ATTR_BRIGHTNESS, light.brightness)
```

Comment:
I think that if the user wants to set an XY color and does not provide a brightness, it will still work for lights that support XY colors. It's just that we can't do a conversion and so can't automatically fix it, I think that is totally ok.